src/notation/view/internal/dynamicpopupmodel.cpp

```diff
@@ -0,0 +1,180 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "dynamicpopupmodel.h"
+
+#include "engraving/dom/dynamic.h"
+#include "engraving/dom/factory.h"
+#include "engraving/types/symnames.h"
+
+#include "log.h"
+
+using namespace mu::notation;
+using namespace mu::engraving;
+
+static const QList<QList<DynamicPopupModel::PageItem> > DYN_POPUP_PAGES = {
+    {   // Page 1
+        { DynamicType::PP,     30, 1.5, DynamicPopupModel::Dynamic },
+        { DynamicType::P,      21, 1.0, DynamicPopupModel::Dynamic },
+        { DynamicType::MP,     29, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::MF,     30, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::F,      23, 2.5, DynamicPopupModel::Dynamic },
+        { DynamicType::FF,     30, 2.5, DynamicPopupModel::Dynamic },
+    },
+    {   // Page 2
+        { DynamicType::FFF,    38, 2.5, DynamicPopupModel::Dynamic },
+        { DynamicType::FFFF,   45, 2.5, DynamicPopupModel::Dynamic },
+        { DynamicType::FFFFF,  53, 2.5, DynamicPopupModel::Dynamic },
+    },
+    {   // Page 3
+        { DynamicType::FP,     30, 2.5, DynamicPopupModel::Dynamic },
+        { DynamicType::PF,     32, 2.0, DynamicPopupModel::Dynamic },
+        { DynamicType::SF,     25, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::SFZ,    29, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::SFF,    33, 0.5, DynamicPopupModel::Dynamic },
+    },
+    {   // Page 4
+        { DynamicType::SFFZ,   37, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::SFP,    33, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::RFZ,    30, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::RF,     26, 0.5, DynamicPopupModel::Dynamic },
+        { DynamicType::FZ,     26, 2.5, DynamicPopupModel::Dynamic },
+    },
+    {   // Page 5
+        { DynamicType::PPPPPP, 74, 2.0, DynamicPopupModel::Dynamic },
+        { DynamicType::FFFFFF, 60, 2.5, DynamicPopupModel::Dynamic },
+    },
+    {   // Page 6
+        { DynamicType::PPPPP,  64, 2.0, DynamicPopupModel::Dynamic },
+        { DynamicType::PPPP,   52, 2.0, DynamicPopupModel::Dynamic },
+        { DynamicType::PPP,    44, 2.0, DynamicPopupModel::Dynamic },
+    },
+    {   // Page 7 - Hairpins
+        { DynamicType::OTHER,  62, 0.0, DynamicPopupModel::Crescendo },
+        { DynamicType::OTHER,  62, 0.0, DynamicPopupModel::Decrescendo },
+    },
+};
+
+DynamicPopupModel::DynamicPopupModel(QObject* parent)
+    : AbstractElementPopupModel(PopupModelType::TYPE_DYNAMIC, parent)
+{
+}
+
+QString DynamicPopupModel::fontFamily() const
+{
+    return QString::fromStdString(m_item->score()->engravingFont()->family());
```

Comment:
You might want to add a safety check here to prevent a nullptr dereference:
```c++
    IF_ASSERT_FAILED(m_item) {
        return QString();
    }
```

Reply:
Done